ambry-store/src/test/java/com.github.ambry.store/BlobStoreStatsTest.java

```diff
@@ -0,0 +1,335 @@
+      String serviceId = expectedContainerValidSizeEntry.getKey();
+      assertTrue("Expected serviceId: " + serviceId + " not found", actualContainerValidSizeMap.containsKey(serviceId));
+      Map<String, Long> innerMap = expectedContainerValidSizeEntry.getValue();
+      for (Map.Entry<String, Long> innerEntry : innerMap.entrySet()) {
+        String containerId = innerEntry.getKey();
+        assertTrue("Expected containerId: " + containerId + " not found in serviceId: " + serviceId,
+            innerMap.containsKey(containerId));
+        long expectedContainerValidSize = innerEntry.getValue();
+        long actualContainerValidSize = actualContainerValidSizeMap.get(serviceId).get(containerId);
+        assertEquals("Valid data size mismatch for serviceId: " + serviceId + " containerId: " + containerId,
+            expectedContainerValidSize, actualContainerValidSize);
+        totalValidSize += expectedContainerValidSize;
+      }
+    }
+
+    return totalValidSize;
+  }
+
+  /**
+   * Verify the correctness of valid data size information per log segment returned by BlobStoreStats for a given {@link TimeRange}.
+   * @param timeRange the {@link TimeRange} to be used for the verification
+   * @return the total valid data size of all log segments
+   */
+  private long verifyLogSegmentValidSize(TimeRange timeRange) throws StoreException {
+    Pair<Long, NavigableMap<String, Long>> actualLogSegmentValidSizeMap = blobStoreStats.getValidDataSizeByLogSegment(timeRange);
+    assertTrue("Valid data size collection time should be in the range",
+        timeRange.getStartTimeInMs() <= actualLogSegmentValidSizeMap.getFirst()
+            && timeRange.getEndTimeInMs() >= actualLogSegmentValidSizeMap.getFirst());
+
+    int expectedNumberOfLogSegments = 0;
+    long expectedTotalLogSegmentValidSize = 0L;
+    LogSegment logSegment = state.log.getFirstSegment();
+
+    while (logSegment != null) {
+      String logSegmentName = logSegment.getName();
+      assertTrue("Log segment: " + logSegmentName + " not found",
+          actualLogSegmentValidSizeMap.getSecond().containsKey(logSegmentName));
+
+      long expectedLogSegmentValidSize =
+          state.getValidDataSizeForLogSegment(logSegment, timeRange.getEndTimeInMs(), state.time.milliseconds());
+      long actualLogSegmentValidSize = actualLogSegmentValidSizeMap.getSecond().get(logSegmentName);
+      assertEquals("Valid data size mismatch for log segment: " + logSegmentName, expectedLogSegmentValidSize,
+          actualLogSegmentValidSize);
+
+      expectedTotalLogSegmentValidSize += expectedLogSegmentValidSize;
+      expectedNumberOfLogSegments++;
+      logSegment = state.log.getNextSegment(logSegment);
+    }
+
+    assertEquals("Mismatch in number of log segments", expectedNumberOfLogSegments,
+        actualLogSegmentValidSizeMap.getSecond().size());
+
+    Pair<Long, Long> actualTotalValidSize = blobStoreStats.getValidSize(timeRange);
```

Comment:
verifyLogSegmentValidSize verifies all log segment related APIs which includes total. Also, we happen to have all the information we need to verify total here, so I thought why introduce another method that does similar things.